server/server.go

```diff
@@ -853,6 +854,45 @@ func (s *Server) v1DataPut(w http.ResponseWriter, r *http.Request) {
 	}
 }
 
+func (s *Server) v1DataDelete(w http.ResponseWriter, r *http.Request) {
+	ctx := r.Context()
+	vars := mux.Vars(r)
+
+	path, ok := storage.ParsePath("/" + strings.Trim(vars["path"], "/"))
+	if !ok {
+		writer.Error(w, http.StatusBadRequest, types.NewErrorV1(types.CodeInvalidParameter, "bad path: %v", vars["path"]))
+		return
+	}
+
+	txn, err := s.store.NewTransaction(ctx, storage.WriteParams)
+	if err != nil {
+		writer.ErrorAuto(w, err)
+		return
+	}
+
+	_, err = s.store.Read(ctx, txn, path)
+
+	if err != nil && !storage.IsNotFound(err) {
+		s.abortAuto(ctx, txn, w, err)
+		return
+	} else if r.Header.Get("If-None-Match") == "*" {
```

Comment:
This branch is not needed. It tells the server to only DELETE if the resource does not exist.